### PR TITLE
V3

### PR DESCRIPTION
**Pull request by moonglum**
This PR introduces a few breaking changes. We collect them to have a nice, clean v3 - dropping some things that drag us down along the way.

* We drop Virtual entirely. It was not used, and adds significant complexity
  * This allows us to collapse the bundle inheritancy hierarchy
* We drop support for custom extensions
  * I don't think we should have even included it in the first place, I think it is a violation of the faucet philosophy of only supporting "standard code"
  * The file extensions for TypeScript & JSX are added automatically
* We make ESM the new default format (previously it was IIFE). It is a breaking change, but a very necessary one.
* Remove an old unused test util and a really old deprecation warning
* We drop support for Node 14

* [x] Use [ES2015 Syntax in generated code](https://github.com/rollup/rollup/pull/4215)

### CODE REVIEW

**Comment by FND** [comment]
FYI: In an effort to refamiliarize myself with this, I've reviewed every commit here and made some [minor changes](https://github.com/faucet-pipeline/faucet-pipeline-js/compare/c97fc66bb1644db3af4edd260617c0ac18326c18..655e79c4743d931caabf76fc7cb7c6eb76e19156) (mostly tweaking and amending commit messages).

We'll now need to update our test suite to account for latest transpilation changes, but there are also dependencies to be updated (it might make sense to do the latter first).

File: lib/bundle/bundler.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ module.exports = function generateBundle(entryPoint, config, cache) {
 			return bundle.generate(writeConfig);
 		}).
 		then(({ output }) => {
+			output = output.filter(item => item.type !== "asset"); // XXX: simplistic?
```
Comment (FND) [inline lib/bundle/bundler.js:17]: I'm not entirely sure this is correct; it's really hard to keep up with Rollup's API. I mostly guessed based on empiricism and [Rollup's change log](https://github.com/rollup/rollup/blob/master/CHANGELOG.md#300).

**Review by FND — COMMENTED** [review]
all dependencies should be up to date now, with corresponding adjustments in our code

File: lib/bundle/bundler.js [hunk 2]
```diff
@@ -6,6 +6,9 @@ module.exports = function generateBundle(entryPoint, config, cache) {
 	let { readConfig, writeConfig } = config;
 	let options = Object.assign({}, readConfig, {
 		input: entryPoint,
+		output: {
+			generatedCode: "es2015"
```
Comment (FND) [inline lib/bundle/bundler.js:10]: @moonglum: This doesn't seem to make any difference, at least I didn't have to adjust any tests? (cf. af36dfb22fd64248db4decc4c39ab9f211b042a6)

File: lib/bundle/bundler.js (hunk 1 above)
Comment (moonglum) [reply lib/bundle/bundler.js:20]: :see_no_evil:

File: lib/bundle/bundler.js (hunk 2 above)
Comment (moonglum) [reply lib/bundle/bundler.js:10]: Yes, it is weird. For me, the option sounds like something we SHOULD set. But it's weird that we can't find any consequences in the code... I would still vote for keeping the option in here, even though it feels weird.

**Review by moonglum — COMMENTED** [review]
Looks good to me :+1: We need to adjust the version numbers in the package.jsons - but apart from that, I would say: Let's release them as 3.0.0 :+1:

File: lib/bundle/bundler.js (hunk 2 above)
Comment (FND) [reply lib/bundle/bundler.js:10]: Yup, fine by me - so ... :shipit: ?

**Pull request merged**